Ability to Force Update Naomi upon auto fail & Uninstall Naomi

AustinCasteel opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
With these changes when you run the installer instead of checking for the profile.yml file to determine if Naomi is already installed or not it now looks for directories. If the ~/.config/naomi dir exists then it will prompt you with a message, if not then you can proceed with the installation process.

The uninstall feature is implemented as a flag for the install script. If you run 'naomi-setup.sh -uninstall' it with ask for a conformation and then proceed to clean Naomi off of the system.

The update check on Naomi startup has also been updated to include a force update. If auto update fails (which happens when the Naomi dir does not match what the repo has tracked, e.i. modified files for developers or installed 3rd party plugins) instead of immediately launching it will ask if you would like to force and update and proceed as such.

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
[PR #318 for Uninstalling](https://github.com/NaomiProject/Naomi/issues/318)
& an issue discussed on [Discord](https://discord.com/channels/320348612813127684/528697897685090304/805147734587867156) about the Auto Update Fail.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
To flesh out the auto update system a little more and implement a way to uninstall Naomi

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

AustinCasteel committed:
Force Update & Uninstall

aaronchantrill reviewed (changes requested):
"uninstall" code never actually runs right now. Try changing the order of the "elif [ -d ~/.config/naomi ]" and "elif [ "$1" == "uninstall" ]" blocks.

AustinCasteel committed:
Swapping order of operations

aaronchantrill committed:
Uninstall

${NAOMI_BASE}/naomi-setup.sh was not passing arguments to
${NAOMI_BASE}/installers/naomi-setup.sh, so passing arguments
to the first script had no effect on the second, which was
expecting the arguments.

I also changed the argument from "-uninstall" or "uninstall" to
"--uninstall" which is a more typical way to pass arguments in
Linux. I also changed the "Continue?" prompt to accept lower
case y/n and used SUDO_COMMAND to give the user a chance to
cancel if they don't want to delete the ~/Naomi or ~/.config/naomi
directories.

AustinCasteel committed:
Merge pull request #16 from aaronchantrill/autoupdate-fallover

Uninstall
